src/webgpu/api/validation/createComputePipeline.spec.ts

```diff
@@ -0,0 +1,164 @@
+export const description = `
+createComputePipeline and createComputePipelineAsync validation tests.
+
+TODO:
+For createComputePipeline and its async version createComputePipelineAsync,
+each start with a valid descriptor, than for the only one compute stage, make
+following errors:
+- compute stage's module is an invalid object
+- compute stage's module is of other stage, i.e., vertex or gragment
+- stage's entryPoint doesn't exist, including following situation:
+  - the entryPoint in module is / isn't 'main'
+  - the assigned name is different from module's name
+  - the assigned name is empty string
+  - the assigned name contain illegal character
+`;
+
+import { params, poptions } from '../../../common/framework/params_builder.js';
+import { makeTestGroup } from '../../../common/framework/test_group.js';
+
+// import { kAllTextureFormats, kAllTextureFormatInfo } from '../../capability_info.js';
+import { ValidationTest } from './validation_test.js';
+
+type TShaderStage = 'compute' | 'vertex' | 'fragment' | 'empty';
+
+class F extends ValidationTest {
+  getShaderModule(
+    shaderStage: TShaderStage = 'compute',
+    entryPoint: string = 'main'
+  ): GPUShaderModule {
+    let code;
+    switch (shaderStage) {
+      case 'compute': {
+        code = `[[stage(compute)]] fn ${entryPoint}() {}`;
```

Comment:
nit: can you add `workgroup_size(1)` too? The group just decided to require these annotations on all compute entrypoints now.